motionplan/check.go

```diff
@@ -181,9 +151,9 @@ func checkPlanRelative(
 	segments = append(segments, &ik.Segment{
 		StartPosition:      currentPoseInWorld.Pose(),
 		EndPosition:        currentArcEndPose,
-		StartConfiguration: checkFrameCurrentInputs,
-		EndConfiguration:   arcEndInputs,
-		Frame:              sf,
+		StartConfiguration: frameCurrentInputs,
+		EndConfiguration:   plan.Trajectory()[wayPointIdx][checkFrame.Name()],
```

Comment:
[nit] for safety this could be  done through plan.Trajectory.GetFrameInputs(frameName)